tests/util/test_logging.py

```diff
@@ -49,6 +51,46 @@ async def test_logging_with_queue_handler() -> None:
     assert simple_queue.empty()
 
 
+@pytest.mark.parametrize(
+    ("version", "exc", "should_filter"),
+    [
+        ("2023.10.0", HomeAssistantError, True),
+        ("2023.10.0b0", HomeAssistantError, False),
+        ("2023.10.0", ConfigError, True),
+        ("2023.10.0b0", ConfigError, False),
+        ("2023.10.0", KeyError, False),
+        ("2023.10.0b0", KeyError, False),
+    ],
+)
+async def test_suppressed_logging_stack_trace(
+    hass: HomeAssistant,
+    caplog: pytest.LogCaptureFixture,
+    version: str,
+    exc: Exception,
+    should_filter: bool,
+) -> None:
+    """Test logging stack trace on stable builds.
+
+    Stack traces for HomeAssistantError should be suppressed for stable builds.
+    """
+    logger = logging.getLogger("")
+    with patch("homeassistant.core.__version__", version):
+        filter = logging_util.SuppressHomeAssistantErrorStackTrace()
+        logger.addFilter(filter)
+
+    try:
+        raise exc("Test exception")
+    except exc:
+        logger.exception(exc)
+
+    logger.removeFilter(filter)
+
+    if should_filter:
+        assert 'raise exc("Test exception")' not in caplog.text
```

Comment:
I'd also assert what is actually logged. There should still be an error log, right?

Reply:
Changed the test a bit to assert the log record.exc_info is asserted to be `None` or to be a `tuple` and to contain a `TracebackType` object.